app/assets/javascripts/admin/views/scripts/script_list_view.js

```diff
@@ -1,48 +1,150 @@
+// @flow
+/* eslint-disable jsx-a11y/href-no-hash */
+
 import _ from "lodash";
-import app from "app";
-import Marionette from "backbone.marionette";
-import ScriptListItemView from "admin/views/scripts/script_list_item_view";
-
-class ScriptListView extends Marionette.CompositeView {
-  static initClass() {
-    this.prototype.className = "container wide";
-    this.prototype.childView = ScriptListItemView;
-    this.prototype.childViewContainer = "tbody";
-  }
+import * as React from "react";
+import { Table, Icon, Spin, Button, Input, Modal } from "antd";
+import Utils from "libs/utils";
+import Request from "libs/request";
+import messages from "messages";
+import type { APIScriptType } from "admin/api_flow_types";
+
+const { Column } = Table;
+const { Search } = Input;
+
+type State = {
+  isLoading: boolean,
+  scripts: Array<APIScriptType>,
+  searchQuery: string,
+};
 
-  template = () =>
-    _.template(`\
-<h3>Scripts</h3>
-<table class="table table-striped table-details">
-  <thead>
-    <tr>
-      <th>#</th>
-      <th>Name</th>
-      <th>Owner</th>
-      <th>Gist URL</th>
-      <th>Actions</th>
-    </tr>
-  </thead>
-  <tbody></tbody>
-</table>\
-`);
-
-  initialize() {
-    this.collection.fetch();
-
-    this.listenTo(app.vent, "paginationView:filter", this.filterBySearch);
-    this.listenTo(app.vent, "paginationView:addElement", this.createNewScript);
+class ScriptListView extends React.PureComponent<{}, State> {
+  state = {
+    isLoading: true,
+    scripts: [],
+    searchQuery: "",
+  };
+
+  componentDidMount() {
+    this.fetchData();
   }
 
-  createNewScript() {
-    return app.router.navigate("/scripts/create", { trigger: true });
+  async fetchData(): Promise<void> {
+    const url = "/api/scripts";
+    const scripts = await Request.receiveJSON(url);
+
+    this.setState({
+      isLoading: false,
+      scripts: scripts.filter(p => p.owner),
```

Comment:
So, scripts without an owner will be ignored? How can a script be ownerless?

Reply:
Ups. Copy and paste error. We had some cases were projects were ownerless because the user/owner reference was invalid. (Possibly the user was deleted). Will fix.